src/Compilers/CSharp/Portable/Binder/LookupResult.cs

```diff
@@ -286,11 +286,32 @@ internal void MergeEqual(SingleLookupResult result)
             {
                 this.SetFrom(result);
             }
+            else if (Kind == LookupResultKind.WrongArity && result.Kind == LookupResultKind.WrongArity)
+            {
+                // When both results are WrongArity, prefer reporting errors about generic types
+                // over non-generic types to provide more helpful error messages
+                if (preferGenericOverNonGeneric(this.SingleSymbolOrDefault, result.Symbol))
```

Comment:
Consider a name like `isNonGenericVersusGeneric` instead, so that the terms in the function name are in the same order as the tests we do on the parameters.

Currently the name reads as if we are trying to decide whether we prefer or not an incoming generic symbol over a non-generic symbol. But that's not really what it's doing.